starboard/shared/posix/socket_receive_multi_msg.h

```diff
@@ -0,0 +1,28 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef STARBOARD_SHARED_POSIX_SOCKET_RECEIVE_MULTI_MSG_H_
+#define STARBOARD_SHARED_POSIX_SOCKET_RECEIVE_MULTI_MSG_H_
+
+namespace starboard {
+namespace shared {
+namespace posix {
+
+const void* GetSocketReceiveMultiMsgApi();
```

Comment:
In trunk / Starboard 17 this should just be `recvmmsg` or are there any other socket APIs used ?

Reply:
Possibly, but there are also data types involved and recvmmsg is a linux API, not a cobalt API. Other OSs have slightly different versions of this function. I kind of lean towards keeping this because the way this is implemented, we can add `UDP_GRO`  support without changing the rest of the code.

Reply:
To make this upstreamable to Chromium though you probably want a clean version written against trunk Chromium with just `recvmmsg` ?

Reply:
Yes, that can be done in `udp_socket_posix.cc` that includes `<sys/socket.h>` where the necessary defines live. I can add the corresponding logic there once we update Starboard enough to use `udp_socket_posix.cc`.